x-pack/plugins/security_solution/common/search_strategy/security_solution/index.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ESQuery } from '../../typed_json';
+import {
+  HostDetailsStrategyResponse,
+  HostOverviewRequestOptions,
+  HostsQueries,
+  HostsRequestOptions,
+  HostsStrategyResponse,
+} from './hosts';
+export * from './hosts';
+export type Maybe<T> = T | null;
+
+export type FactoryQueryTypes = HostsQueries;
+
+export interface Inspect {
+  dsl: string[];
+  response: string[];
+}
+
+export interface PageInfoPaginated {
+  activePage: number;
+  fakeTotalCount: number;
+  showMorePagesIndicator: boolean;
+}
+
+export interface CursorType {
+  value?: Maybe<string>;
+  tiebreaker?: Maybe<string>;
+}
+
+export enum Direction {
+  asc = 'asc',
+  desc = 'desc',
+}
+
+export interface SortField {
+  field: string;
+  direction: Direction;
+}
+
+export interface TimerangeInput {
+  /** The interval string to use for last bucket. The format is '{value}{unit}'. For example '5m' would return the metrics for the last 5 minutes of the timespan. */
+  interval: string;
+  /** The end of the timerange */
+  to: string;
+  /** The beginning of the timerange */
+  from: string;
+}
+
+export interface PaginationInput {
+  /** The limit parameter allows you to configure the maximum amount of items to be returned */
+  limit: number;
+  /** The cursor parameter defines the next result you want to fetch */
+  cursor?: Maybe<string>;
+  /** The tiebreaker parameter allow to be more precise to fetch the next item */
+  tiebreaker?: Maybe<string>;
+}
+
+export interface PaginationInputPaginated {
+  /** The activePage parameter defines the page of results you want to fetch */
+  activePage: number;
+  /** The cursorStart parameter defines the start of the results to be displayed */
+  cursorStart: number;
+  /** The fakePossibleCount parameter determines the total count in order to show 5 additional pages */
+  fakePossibleCount: number;
+  /** The querySize parameter is the number of items to be returned */
+  querySize: number;
+}
+
+export interface DocValueFields {
+  field: string;
+  format: string;
+}
+
+export interface RequestBasicOptions {
+  timerange: TimerangeInput;
+  filterQuery: ESQuery | string | undefined;
+  defaultIndex: string[];
+  docValueFields?: DocValueFields[];
+  factoryQueryType?: FactoryQueryTypes;
+}
+
+export interface RequestOptions extends RequestBasicOptions {
+  pagination: PaginationInput;
+  sortField?: SortField;
+}
+
+export interface RequestOptionsPaginated extends RequestBasicOptions {
+  pagination: PaginationInputPaginated;
+  sortField?: SortField;
+}
+
+export type StrategyResponseType<T extends FactoryQueryTypes> = T extends 'host_all'
+  ? HostsStrategyResponse
+  : T extends 'host_details'
+  ? HostDetailsStrategyResponse
```

Comment:
Please consider rename `HostDetailsStrategyResponse ` into `HostOverviewRequestResponse`.
I think the response of host details would be more generic and may include more info than HostOverviewRequestResponse/

Reply:
Not a big fan of the Overview naming but It makes sense to keep it the same to not confuse people so I will change it back.